insights/client/constants.py

```diff
@@ -10,9 +8,8 @@ class InsightsConstants(object):
         os.path.dirname(os.path.abspath(__file__)))
     sleep_time = 180
     command_blacklist = ('rm', 'kill', 'reboot', 'shutdown')
-    default_conf_dir = '/etc/insights-client'
+    default_conf_dir = os.getenv('INSIGHTS_CONF_DIR', default='/etc/insights-client')
```

Comment:
I assume you added this for dev purposes?

Reply:
Yes, it makes running modules in-tree easier, and *shouldn't* have any adverse effects otherwise.

Reply:
What does it mean by running modules in-tree? Do you mean running stuff with `python insights/client/something.py`?

Reply:
Do you think this variable can be passed in by insights-client when spawning the subprocess? It would allow to have dynamic paths for the Client being installed in other place than root.

Reply:
> Do you mean running stuff with python insights/client/something.py?

Yes, that is correct. It also allows you to more easily spawn the client code as non-root under certain circumstances.

> Do you think this variable can be passed in by insights-client when spawning the subprocess? It would allow to have dynamic paths for the Client being installed in other place than root.

We could look into that. Are you thinking of changing the code that spawns `run.py` to inject that variable into the environment?

Reply:
Actually, that could be pretty neat. At "compile" time, we generate a file similar to the classic `config.h` (`config.py`?) that contains all the typical path values set by `./configure`. Then the client execution code that runs within `/usr/bin/insights-client` will always run with the compiled values. It takes those values and sets them as environment variables when it creates the `run.py` subprocess, allowing the insights-core code to be entirely dynamic and relocatable at runtime.